sher-gil/src/main/java/com/kinnerapriyap/sugar/ShergilViewModel.kt

```diff
@@ -116,11 +118,27 @@ class ShergilViewModel(application: Application) : AndroidViewModel(application)
     fun getCurrentMediaCursor(bucketDisplayName: String? = null): Cursor? =
         MediaGalleryCursorWrapper(cursor.value, bucketDisplayName)
 
-    fun fetchAlbumCursor(): Cursor? =
-        mediaGalleryHandler.fetchAlbum(cursor.value, choiceSpec.allowCamera)
-
-    fun closeCursor() {
-        cursor.value?.close()
+    fun fetchAlbums(): List<MediaGalleryAlbum> {
+        val addedNamesCount: MutableMap<String, Int> = mutableMapOf()
+        val cursor = cursor.value ?: return emptyList()
+        cursor.moveToFirst()
+        while (!cursor.isAfterLast) {
+            val name =
+                cursor.getString(cursor.getColumnIndex(MediaGalleryHandler.BUCKET_DISPLAY_NAME))
```

Comment:
is it possible for this to throw exceptions?

Reply:
Added check for -1, which looks like the only way for exceptions to occur here.